processor/src/test/java/org/mapstruct/ap/test/bugs/_3806/Issue3806Test.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.bugs._3806;
+
+import org.mapstruct.ap.testutil.ProcessorTest;
+import org.mapstruct.ap.testutil.WithClasses;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@WithClasses({ DestinationType.class, SourceType.class, Issue3806Mapper.class })
+public class Issue3806Test {
+
+    @ProcessorTest
+    public void shouldNotClearGetterOnlyCollectionsInUpdateMapping() {
```

Comment:
Can we perhaps add a positive test here as well? e.g. the `source` should have some data, so we want to show that it gets properly populated.

Reply:
sure.